### ISSUE TITLE
[BUG] Some accented lowercase letters cannot be converted to uppercase

### ISSUE THREAD

**Issue by JapanYoshi** [TeamRizu/OutFox#121]
**Please give System Info**

OS: Windows 10

CPU: Intel i7

GPU: Nvidia GeForce

Storage: HDD + SSD

**Please give your StepMania Info**

StepMania Version: 4.8.5
Theme: SoundWaves

**Describe the problem**

Latin characters outside the Windows-1252 code page appear lowercase when converted to all-caps text. For example, some strings in the Polish version contain the characters ąćęłńśźż, which suffer from this error (e.g. “dołącz” (join) → DOłąCZ)

**Describe what should happen**

All Unicode characters should be capitalized.

**Logs and Additional context**

1. Go into the game folder
2. Open the Logs folder
3. Grab every file in there and post with the issue.
4. Go back to the game folder, if there's an crashinfo.txt then send it too.

Add any other context or screenshots here.

**Comment by Scraticus** [comment]
This is a bug with how `string.upper` / string libs work within Lua itself. To fix it we have just removed the upper calls in soundwaves, which will be released with 4.9.0s hotfix.

If you want caps in future, make them caps in the .ini files.

**Comment by JapanYoshi** [comment]
> If you want caps in future, make them caps in the .ini files.

I disagree strongly with this notion. All caps text is a stylistic choice, not word/localization choice, and should be done in the frontend; source text should be in the most natural capitalization (sentence case).

**Comment by JapanYoshi** [comment]
> If you want caps in future, make them caps in the .ini files.

I disagree strongly with this notion. All caps text is a stylistic choice, not word/localization choice, and should be done in the frontend; source text should be in the most natural capitalization (sentence case).

**Comment by Scraticus** [comment]
We need to look into the upper case for UTF capable strings. 

For this alpha, it's removed because people are complaining about mixed case localisation you can't have it your own way all the time, sometimes to resolve something, other sacrifices need to be made temporarily.

right now string.upper breaks a lot of the new glyph support we added for the next alpha so it's been taken out until a new solution is found, so we don't end up with lots of issues like this one, where people get upset because of less than half a dozen glyphs not working correctly.

**Comment by JapanYoshi** [comment]
I think we both need to settle down and regain our bearings.

First of all, I'm in no way blaming you for the ancient code you are tasked to maintain.

Notwithstanding, even if this error seems to only affect 8 letters, those letters are necessary for supporting a specific language, and they are, in fact, just the tip of the iceberg. Imagine fixing just these letters, then some time down the line, someone decides to translate this into something with lots of non-Windows-1252 letters like Vietnamese or Azeri.

An external Lua package that supports the whole of Unicode (as it stands now) is probably the best solution in my humble opinion. Capitalizing Unicode text is such a basic thing that it should be supported by default, but that means someone else has probably solved it already.

**Comment by Scraticus** [comment]
we do not use 1252 in outfox as of the next alpha. it's gone.

**Comment by JapanYoshi** [comment]
Good news! Hopefully the to-uppercase function in the new string format is more robust.

**Comment by Scraticus** [comment]
we need to add it to Lua, as i said, it was an unforeseen issue with us adding all the new glyph support and removing the ascii/1252 era pages. They are _supported_ but only as a last resort.

This alpha will fix the mixed case, but have no caps as doUpper() and string.upper are for ASCII/Unicode only, not UTF-8.

Hope this clears up what I meant earlier.

**Comment by Scraticus** [comment]
Right, in testing, we now have a new system to handle the UTF side of outfox. Previous themes and such will use string.upper without issue, but will suffer the downside of the legacy font system.

To display uppercase text - done on the troublesome polish characters:
`Text=GAMESTATE:ToUpper(Screen.String("Start To Join"))` produces:

![2020-10-06_134857](https://user-images.githubusercontent.com/11047768/95259274-d88b9700-081e-11eb-83d2-04c3c6c387ec.jpg)

We even can do a proper UTF friendly tolower() as well:
`Text=GAMESTATE:ToLower(Screen.String("Start To Join"))` 
![2020-10-06_135039](https://user-images.githubusercontent.com/11047768/95259381-ffe26400-081e-11eb-8272-3e18e708efac.jpg)

And as a test: 
Text=GAMESTATE:ToLower(GAMESTATE:ToUpper(Screen.String("Start To Join")))
works as well.

(Thanks to @MrThatKid for helping me test this out!)

Basically, Lua itself doesn't have this functionality, we just added it in the engine so we can put this to bed.

**Comment by Scraticus** [comment]
With the last update making the command a global, we can now have fully localised versions of OutFox in the future now.
